pkg/v3/postprocessors/ineligible_test.go

```diff
@@ -0,0 +1,114 @@
+package postprocessors
+
+import (
+	"bytes"
+	"context"
+	"errors"
+	"log"
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+
+	"github.com/smartcontractkit/ocr2keepers/pkg/v3/types"
+)
+
+func TestNewIneligiblePostProcessor(t *testing.T) {
+	for _, tc := range []struct {
+		name         string
+		stateUpdater types.UpkeepStateUpdater
+		results      []types.CheckResult
+		wantLog      string
+		expectsErr   bool
+		wantErr      error
+	}{
+		{
+			name: "upkeep state is updated as per one check result",
+			stateUpdater: &mockStateUpdater{
+				SetUpkeepStateFn: func(ctx context.Context, result types.CheckResult, state types.UpkeepState) error {
+					return nil
+				},
+			},
+			results: []types.CheckResult{
+				{
+					PipelineExecutionState: 0,
+					Eligible:               false,
+				},
+			},
+			wantLog: "post-processing 1 results, 0 ineligible",
```

Comment:
should the log here indicate 1 ineligible?

Reply:
ah i think there's a bug in ineligible processor when counting ineligible count

Reply:
Should the ineligible loop then become:

```if res.PipelineExecutionState == 0 && !res.Eligible {
	err := p.stateUpdater.SetUpkeepState(ctx, res, ocr2keepers.Ineligible)
	if err != nil {
		merr = errors.Join(merr, err)
		continue
	}
	ineligible++
}```

Reply:
yup i think so!